src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Window.cs

```diff
@@ -6304,12 +6305,19 @@ private void ApplyTaskbarItemInfo()
                 try
                 {
                     taskbarList = (ITaskbarList)Activator.CreateInstance(Type.GetTypeFromCLSID(new Guid(CLSID.TaskbarList)));
-                    taskbarList.HrInit();
+
+                    hr = taskbarList.HrInit();
+                    if (hr != HRESULT.S_OK)
+                    {
+                        // Taskbar not available (no user logged in, running under terminal service, custom shell, etc.)
+                        HandleTaskbarListError(hr);
+                        return;
+                    }
```

Comment:
What about releasing the COM?
![image](https://user-images.githubusercontent.com/4403806/170179436-fd367d1b-b61a-4827-a781-34ceee6d4205.png)

Reply:
Thank you @RussKie, as noted in the description, the interface is released in the finally block below.

Reply:
Ah, I totally missed it. Thank you.